crates/oxc_linter/src/lib.rs

```diff
@@ -90,6 +115,44 @@ impl Linter {
             }
         }
 
+        let adapter = LintAdapter { semantic: semantic.clone() };
+        let la = Arc::from(adapter);
+        for input_query in TRUSTFALL_RULES.iter() {
+            for data_item in execute_query(
+                &SCHEMA,
+                Arc::clone(&la),
+                input_query.query.as_str(),
+                input_query.args.clone(),
+            )
+            .expect(
+                format!("not a legal query in query: \n\n\n{}", input_query.query.as_str())
+                    .as_str(),
+            )
+            .take(usize::MAX)
+            {
+                let transparent: BTreeMap<_, TransparentValue> =
+                    data_item.into_iter().map(|(k, v)| (k, v.into())).collect();
+                // println!("\n{}", serde_json::to_string_pretty(&transparent).unwrap());
+                ctx.with_rule_name("a rule");
+                let TransparentValue::Uint64(start) = &transparent["span_start"] else {
+                        println!("{:?}", transparent);
+                        unreachable!()
+                    };
+                let TransparentValue::Uint64(end) = &transparent["span_end"] else {
+                        println!("{:?}", transparent);
+                        unreachable!()
+                    };
```

Comment:
You should be able to get something more ergonomic here by using `TryIntoStruct`, new in Trustfall v0.5:
https://github.com/obi1kenobi/trustfall/releases/tag/trustfall-v0.5.0

Recall that serde ignores extra data that isn't relevant to the fields it's trying to deserialize, so you should be able to define a struct like:
```rust
#[derive(Debug, serde::Deserialize)]
struct SpanInfo {
    span_start: u64,
    span_end: u64,
}
```
and then use `TryIntoStruct` to avoid all the manual deserialization you're doing here.

Unrelated: `unreachable!()` takes format-like parameters: `unreachable!("{transparent:?}")`

Reply:
Done, thanks!